Sources/ClientRuntime/Client/ClientBuilder.swift

```diff
@@ -4,29 +4,36 @@
 //
 // SPDX-License-Identifier: Apache-2.0
 //
+
 public class ClientBuilder<ClientType: Client> {
 
-    private var plugins: [Plugin]
+    private struct PluginContainer: Plugin {
+        let plugin: any Plugin<ClientType.Config>
 
-    public init(defaultPlugins: [Plugin] = []) {
-        self.plugins = defaultPlugins
+        func configureClient(clientConfiguration: inout ClientType.Config) async throws {
+            try await plugin.configureClient(clientConfiguration: &clientConfiguration)
+        }
     }
```

Comment:
The `PluginContainer` class above is used to type-erase the actual plugin so that multiple plugins may be stored in a collection together.